diff.go

```diff
@@ -46,15 +47,31 @@ type diffRemote struct {
 }
 
 func (r *diffRemote) Apply(ctx context.Context, desc ocispec.Descriptor, mounts []mount.Mount, opts ...diff.ApplyOpt) (ocispec.Descriptor, error) {
-	var config diff.ApplyConfig
+	var (
+		config           diff.ApplyConfig
+		dcparametersjson []byte
+		err              error
+	)
+
 	for _, opt := range opts {
 		if err := opt(&config); err != nil {
 			return ocispec.Descriptor{}, err
 		}
 	}
+
+	if config.CryptoConfig.DecryptConfig != nil && len(config.CryptoConfig.DecryptConfig.Parameters) > 0 {
+		dcparametersjson, err = json.Marshal(config.CryptoConfig.DecryptConfig.Parameters)
```

Comment:
hmm... Is it possible to just marshal the entire `DecryptConfig` object, and unmarshal it on the other side?

Reply:
The explicit marshal is due to the map[string][][]byte that the proto buf language doesn't handle. It is possible to do what you are saying using `json.Marshal()` though it feels like it's defeating the purpose of the proto buffer. If ever DecryptConfig was extended with other fields, I am not sure what the better way to handle this would be, but the existing pattern seems to be to add new fields to the protobuf 'message'. Maybe there should be a new type with a constructor that converts the DecryptConfig into a DecryptConfigPB where the json conversion of the Parameters field is handled, though with the single field that we have right now it doesn't seem necessary.